text/src/swam/witx/unresolved/Types.scala

```diff
@@ -0,0 +1,43 @@
+package swam
+package witx
+package unresolved
+
+import swam.witx.traverser.TypesTraverser
+
+/**
+  *@author Javier Cabrera-Arteaga on 2020-03-18
+  */
+abstract class BaseWitxType(val tpeName: String, val size: Int = 4, val pad: Int = 0)
+
+abstract class BasicType(name: String, override val size: Int, override val pad: Int = 0)
+    extends BaseWitxType(name, size, pad)
```

Comment:
Also please don't use abstract classes as AST base classes if they are just interfaces, especially if this is to have default values. Avoid this, prefer:
```suggestion
sealed trait BaseWitxType {
  val tpeName: String
  val size: Int
  val pad: Int
}
  
sealed trait BasicType extends BaseWitxType {
  val name: String
}
```